main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/mux"
+)
+
+//===========================================================================================
+//===========================================================================================
+//MODEL
+var db *sql.DB
+var err error
+
+/*create table users(Username varchar(10) primary,
+FirstName varchar(10),LastName varchar(10),Password varchar(20),
+Gender varchar(10),Country varchar(20),Age varchar(10)); */
+
+//User struct
+type User struct {
+	Username  string `json:"username"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Password  string `json:"password"`
+	Gender    string `json:"gender"`
+	Country   string `json:"country"`
+	Age       string `json:"age"`
+	Email     string `json:"email"`
+}
+
+//===========================================================================================
+//CONTROLLERSb
+func handleSignUp(w http.ResponseWriter, r *http.Request) {

```

Comment:
In golang controllers are called handlers, as you may know already. Create a separate package called handlers there.